src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp

```diff
@@ -75,26 +75,53 @@ QVariant QgsValueRelationWidgetWrapper::value() const
       }
     }
 
-    QVariantList vl;
-    //store as QVariantList because it's json
-    for ( const QString &s : qgis::as_const( selection ) )
+    if ( layer()->fields().at( fieldIdx() ).type() == QVariant::Map )
+    {
+      QVariantList vl;
+      //store as QVariantList because the field type supports data structure
+      for ( const QString &s : qgis::as_const( selection ) )
+      {
+        // Convert to proper type
+        const QVariant::Type type { fkType() };
+        switch ( type )
+        {
+          case QVariant::Type::Int:
+            vl.push_back( s.toInt() );
+            break;
+          case QVariant::Type::LongLong:
+            vl.push_back( s.toLongLong() );
+            break;
+          default:
+            vl.push_back( s );
+            break;
+        }
+      }
+      v = vl;
+    }
+    else
     {
-      // Convert to proper type
-      const QVariant::Type type { fkType() };
-      switch ( type )
+      QStringList sl;
+      for ( const QString &s : qgis::as_const( selection ) )
       {
-        case QVariant::Type::Int:
-          vl.push_back( s.toInt() );
-          break;
-        case QVariant::Type::LongLong:
-          vl.push_back( s.toLongLong() );
-          break;
-        default:
-          vl.push_back( s );
-          break;
+        // Convert to proper type
+        const QVariant::Type type { fkType() };
+        switch ( type )
+        {
+          case QVariant::Type::Int:
+          case QVariant::Type::LongLong:
+            sl.push_back( s );
+            break;
+          default:
+            QString newS = s;
+            newS.replace( '\\', QStringLiteral( "\\\\" ) );
```

Comment:
Please use raw string here, much more readable than counting slashes.
```suggestion
            newS.replace( '\\', QStringLiteral( R"(\\)" ) );
```